decoder.go

```diff
@@ -472,7 +462,9 @@ func (d *decoder) textUnmarshaller(key string, value reflect.Value, str string)
 		return nil
 	}
 
-	if value.IsNil() {
+	kind := value.Kind()
+
+	if kind >= reflect.Array && kind <= reflect.Slice && value.IsNil() {
```

Comment:
i don't know what types are in between the Array and Slice, is there a way to be more explicit here?